arwen/ethapi/ethereumei.go

```diff
@@ -710,12 +711,12 @@ func ethreturnDataCopy(context unsafe.Pointer, resultOffset int32, dataOffset in
 	}
 
 	returnDataSlice, err := arwen.GuardedGetBytesSlice(ethReturnData, dataOffset, length)
```

Comment:
Just a question:

```
if int32(len(ethReturnData)) < dataOffset+length {
	return
}
```

Is this desired behavior or a silent error? If it weren't this return here, then the subsequent `arwen.GuardedGetBytesSlice` would have resulted in error (bad bounds), then fault.

Don't we actually want to fault here instead of hiding the error and return to the caller and don't even tell him that the copy didn't work? @sasurobert

Reply:
good catch. We should have an error here.

Reply:
Indeed. Added a call to `arwen.WithFault()` before return.